SU2_CFD/src/variables/CDiscAdjVariable.cpp

```diff
@@ -37,143 +37,49 @@
-    DualTime_Derivative_n = new su2double[nVar];
+    DualTime_Derivative.resize(nPoint,nVar) = su2double(0.0);
+    DualTime_Derivative_n.resize(nPoint,nVar) = su2double(0.0);
+
+    Solution_time_n.resize(nPoint,nVar) = su2double(0.0);
+    Solution_time_n1.resize(nPoint,nVar) = su2double(0.0);
   }
 
-  Solution_Direct = new su2double[nVar];
-  External        = new su2double[nVar];
-  External_Old    = new su2double[nVar];
+  Solution_Direct.resize(nPoint,nVar);
+  Sensitivity.resize(nPoint,nDim) = su2double(0.0);
 
-  Sensitivity = new su2double[nDim];
+  for (unsigned long iPoint = 0; iPoint < nPoint; ++iPoint)
+    for (unsigned long iVar = 0; iVar < nVar; ++iVar)
+      Solution(iPoint,iVar) = sol[iVar];
 
-  unsigned short iVar,iDim;
+  External = Solution;
+  External_Old.resize(nPoint,nVar) = su2double(0.0);
 
-  for (iDim = 0; iDim < nDim; iDim++) {
-    Sensitivity[iDim] = 0.0;
-  }
+  if (fsi) {
+    Geometry_Direct.resize(nPoint,nDim) = su2double(0.0);
+    Solution_Geometry.resize(nPoint,nDim) = su2double(1e-16);
+    Solution_Geometry_Old.resize(nPoint,nDim) = su2double(0.0);
+    Cross_Term_Derivative.resize(nPoint,nVar) = su2double(0.0);
+    Geometry_CrossTerm_Derivative.resize(nPoint,nDim) = su2double(0.0);
+    Geometry_CrossTerm_Derivative_Flow.resize(nPoint,nDim) = su2double(0.0);
 
-  for (iVar = 0; iVar < nVar; iVar++) {
-    Solution[iVar]      = val_solution[iVar];
-    External[iVar]      = val_solution[iVar];
+    Solution_BGS.resize(nPoint,nVar) = su2double(0.0);
+    Solution_Geometry_BGS_k.resize(nPoint,nDim) = su2double(0.0);
   }
 
-  if (dual_time) {
-    for (iVar = 0; iVar < nVar; iVar++) {
-      Solution_time_n[iVar]  = 0.0;
-      Solution_time_n1[iVar] = 0.0;
-      DualTime_Derivative[iVar] = 0.0;
-      DualTime_Derivative_n[iVar] = 0.0;
-    }
-  }
-
-  if (fsi){
-    Solution_Geometry       = new su2double[nDim];
-    Geometry_Direct         = new su2double[nDim];
-    Solution_Geometry_Old   = new su2double[nDim];
-    Geometry_CrossTerm_Derivative = new su2double[nDim];
-    Geometry_CrossTerm_Derivative_Flow = new su2double[nDim];
-    Cross_Term_Derivative   = new su2double[nVar];
-    Solution_BGS            = new su2double[nVar];
-    Solution_Geometry_BGS_k = new su2double[nDim];
-    for (iDim = 0; iDim < nDim; iDim++) {
-      Geometry_Direct[iDim]       = 0.0;
-      Solution_Geometry[iDim]     = 1e-16;
-      Solution_Geometry_Old[iDim] = 0.0;
-      Solution_Geometry_BGS_k[iDim] = 0.0;
-      Geometry_CrossTerm_Derivative[iDim] = 0.0;
-      Geometry_CrossTerm_Derivative_Flow[iDim] = 0.0;
-    }
-    for (iVar = 0; iVar < nVar; iVar++) {
-      Cross_Term_Derivative[iVar] = 0.0;
-      Solution_BGS[iVar]          = 0.0;
-    }
-  }
-  
   if (config->GetMultizone_Problem()) {
-
-    Solution_BGS            = new su2double[nVar];
-    for (iVar = 0; iVar < nVar; iVar++) {
-      Solution_BGS[iVar]          = 0.0;
-    }
-
+    Solution_BGS.resize(nPoint,nVar) = su2double(0.0);
```

Comment:
> I fixed 3 uninitialized warnings, maybe those are the ones you are referring to?

Yes, and this one has been the only critical one ;)

Reply:
I also needed Solution_Old

Reply:
Ok.. it should have had an influence on the residual computation only (it's fixed in the various fixes branch), as it's not used for the solution algorithm. Anyway, might suffice for a crash :p